metadata/metadata_test.go

```diff
@@ -92,6 +92,397 @@ var _ = Describe("Metadata", func() {
 		})
 	})
 
+	When("CopyAnnotationsByPrefix is called", func() {
+		It("should return all the annotations matching the given prefix", func() {
+			podSource := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{"foo": "bar", "baz": "qux"},
+				},
+			}
+			podDest := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{"quux": "corge"},
+				},
+			}
+
+			CopyAnnotationsByPrefix(podSource, podDest, "fo")
```

Comment:
This returns an error. It has to be checked.
```suggestion
			Expect(CopyAnnotationsByPrefix(podSource, podDest, "fo")).To(Succeed())
```

Reply:
Added, thanks!